cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -208,7 +217,12 @@ public Future<PreviewRequest> startPreview(PreviewRequest previewRequest) throws
     }
 
     LOG.debug("Starting preview for {}", programId);
-    ProgramController controller = programLifecycleService.start(programId, userProps, false, true);
+    ProgramStartRequest startRequest = programLifecycleService.prepareProgramStart(programId, userProps, false, true);
+    programStateWriter.start(startRequest.getProgramRunId(), startRequest.getProgramOptions(), null,
```

Comment:
Consider doing it inside ProgramLifecycleService.  The callers don't need to know about this topic, less classes care about it, the better

Reply:
Done.